src/details/ArborX_Ray.hpp

```diff
@@ -273,6 +274,45 @@ KOKKOS_INLINE_FUNCTION float overlapDistance(Ray const &ray,
   return (tmax - tmin);
 }
 
+// Möller–Trumbore intersection algorithm
```

Comment:
I found a pretty nice explanation of the algorithm [here](https://www.scratchapixel.com/lessons/3d-basic-rendering/ray-tracing-rendering-a-triangle/moller-trumbore-ray-triangle-intersection). If you scroll down, they have their source code which uses better names for the variable.